resources/aws/asg_stack.go

```diff
@@ -194,11 +201,31 @@ func (s *ASGStack) Delete() error {
 	return nil
 }
 
+// CheckIfExists checks if there is an autoscaling group stack in Cloud Formation
+// with the provided name.
+func (s *ASGStack) CheckIfExists() (bool, error) {
+	_, err := s.findExisting()
+	if IsNotFound(err) {
+		return false, nil
+	} else if err != nil {
+		return false, microerror.Mask(err)
+	}
+
+	return true, nil
+}
+
 func (s *ASGStack) findExisting() (*cloudformation.Stack, error) {
 	stacks, err := s.Client.DescribeStacks(&cloudformation.DescribeStacksInput{
 		StackName: aws.String(s.Name),
 	})
 	if err != nil {
+		underlying := microerror.Cause(err)
+		if awserr, ok := underlying.(awserr.Error); ok {
+			if awserr.Message() == fmt.Sprintf(stackDoesNotExistError, s.Name) {
```

Comment:
This is the error handling with the string comparison. Unfortunately I can't find a cleaner way to do it.